server/src/main/java/app/user/UserController.java

```diff
@@ -17,27 +20,14 @@ public class UserController {
     private final UserService userService;
 
     @GetMapping("/{userId}/idprofile")
-    public ResponseEntity<UserProfileResponse> getUserProfile(@PathVariable Integer userId, Authentication authentication) {
-        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
-        if (!userDetails.getAuthorities().stream()
-                .anyMatch(auth -> auth.getAuthority().equals("ADMIN") || auth.getAuthority().equals("USER"))) {
-            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-        }
-
-        User user = userService.getUserById(userId);
-        UserProfileResponse response = UserProfileResponse.builder()
-                .id(user.getId())
-                .name(user.getName())
-                .email(user.getEmail())
-                .imageUrl(user.getImageUrl())
-                .dateJoined(user.getDateJoined())
-                .build();
-        return ResponseEntity.ok(response);
+    @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    public ResponseEntity<UserProfileResponse> getUserProfile(@PathVariable Integer userId, @AuthenticationPrincipal UserDetails userDetails) throws AccessDeniedException {

```

Comment:
Zwracaj nie ResponseEntity a odrazu UserProfileResponse 

Jak to działa - jeśłi zwracasz cokolwiek to status jest 200 - OK 

Jeśli chcesz zwrócić coś poza 200 , to zamist tego rzuć błąd i zmapuj go na zwrotkę

Reply:
W zwracaniu ResponseEntity nie jest niczego złego, tylko to trochę zbędne